From 8f5d190292222c0ca2a7b4c2f873140efc586a8e Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Sun, 8 Jan 2017 17:24:18 +0200
Subject: [PATCH 0716/1345] fix: net: mvpp2x: free data_size in error case
 during TXQ init

commit  0904ab37b57780b6aaab013b42fe2fd9f59540ce from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Issue:
- In case of TXQ init error memory allocated for data_size, wasn't
  released.

Change-Id: Id5928211963be7d1e8a103c8d8a7a739b67c597d
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35384
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index f2ebb83..41576e8 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -1310,6 +1310,7 @@ static int mv_pp2x_txq_init(struct mv_pp2x_port *port,
 		txq_pcpu = per_cpu_ptr(txq->pcpu, cpu);
 		kfree(txq_pcpu->tx_skb);
 		kfree(txq_pcpu->tx_buffs);
+		kfree(txq_pcpu->data_size);
 	}
 
 	dma_free_coherent(port->dev->dev.parent,
-- 
1.7.9.5

